### PR TITLE
Upload 3DImage

### LINKED ISSUE #27: Load 3D images
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
The app should be able to load a 3D Tiff images

**Describe alternatives you've considered**

**Additional context**
We'll start by supporting imageJ internal structure

### PR DESCRIPTION
**Pull request by ChrisBaidoo**
# Description

A feature to allow users to upload 3D images. For now we support the upload of .tiff image file type.
closes  #27

# Dependency changes

We have added these dependencies. 

-    "@types/utif": "^3.0.1",
-   "@types/uuid": "^8.3.0"
-    "utif": "^3.1.0",
-    "uuid": "^8.3.2"

# Testing

# Documentation

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by cooper667** [comment]
@ChrisBaidoo you need to update your lcoal `npm` to the latest version so it uses the new lockfile format (I realised that's why my package-lock files were changing on my PRs)

**Comment by ChasNelson1990** [comment]
> Azure Static Web Apps: Your stage site is ready! Visit it here: https://black-wave-0f4ddd903-69.westeurope.azurestaticapps.net

How long do these last? Whenever i click one its blank?

**Comment by philipjackson** [comment]
Does anyone (i.e. @ChasNelson1990 , @cooper667 ) know why the GH actions are failing on `npm ci`?

**Comment by cooper667** [comment]
Fixed, bad package-lock file (you all still need to update npm to v7+)

**Comment by philipjackson** [comment]
> Fixed, bad package-lock file (you all still need to update npm to v7+)

Weird, my npm version is 7.7.5, I updated and re-built package-lock.json just before my last commit, which was the whole point (could it be that I used `npm i` instead of `npm ci` after deleting package-lock.json?)

File: src/ImageFileInfo.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    this.fileName = newFileName;
+  };
+
+  public setFileID = (newFileID: string): void => {
+    this.fileName = newFileID;
+  };
+
+  public getNumOfSlices = (): number => {
+    return this.slicesData.length;
+  };
```
Comment (cooper667) [inline src/ImageFileInfo.ts:42]: These don't seem used?

File: src/toolboxes/background/Canvas.tsx [hunk 2]
```diff
@@ -2,32 +2,34 @@ import React, { Component, ReactNode } from "react";
 
 import { BaseCanvas, CanvasProps as BaseProps } from "@/baseCanvas";
 import drawImageOnCanvas from "./drawImage";
+import ImageFileInfo from "../../ImageFileInfo";
```
Comment (cooper667) [inline src/toolboxes/background/Canvas.tsx:5]: `import ImageFileInfo from "@/ImageFileInfo";`

**Review by cooper667 — APPROVED** [review]
Currently no way the switch slices?

Works with my test one, just a few minor lint fixes (which you _should_ be able to set VSCode up to do on save? (maybe one for @ChasNelson1990 ))

**Comment by cooper667** [comment]
> > Fixed, bad package-lock file (you all still need to update npm to v7+)
> 
> Weird, my npm version is 7.7.5, I updated and re-built package-lock.json just before my last commit, which was the whole point (could it be that I used `npm i` instead of `npm ci` after deleting package-lock.json?)

Yea that's odd, I just deleted the lock and recreated and it worked

**Comment by ChasNelson1990** [comment]
Lets discuss this tommorow. I have a conflicting change in my magic wand branch that i think is the better way to deal with the background canvas but we should chat it out to see?

**Comment by ChasNelson1990** [comment]
Will wait for you guys to move to using ImageData actually?

**Review by ChasNelson1990 — DISMISSED** [review]

**Review by ChasNelson1990 — APPROVED** [review]

**Review by philipjackson — APPROVED** [review]

**Pull request merged**